conversion/conversion.go

```diff
@@ -538,32 +538,56 @@ func getInstances(project string) ([]string, error) {
 
 // WriteSchemaFile writes DDL statements in a file. It includes CREATE TABLE
 // statements and ALTER TABLE statements to add foreign keys.
+// The parameter name should end with a .txt.
 func WriteSchemaFile(conv *internal.Conv, now time.Time, name string, out *os.File) {
 	f, err := os.Create(name)
 	if err != nil {
 		fmt.Fprintf(out, "Can't create schema file %s: %v\n", name, err)
 		return
 	}
-	// The schema file we write out includes comments, includes foreign keys
-	// and doesn't add backticks around table and column names. This file is
-	// intended for explanatory and documentation purposes, and is not strictly
-	// legal Cloud Spanner DDL (Cloud Spanner doesn't currently support comments).
-	// Change 'Comments' to false and 'ProtectIds' to true to write out a
-	// schema file that is legal Cloud Spanner DDL.
-	ddl := conv.SpSchema.GetDDL(ddl.Config{Comments: true, ProtectIds: false, Tables: true, ForeignKeys: true})
-	if len(ddl) == 0 {
-		ddl = []string{"\n-- Schema is empty -- no tables found\n"}
+	spddl := conv.SpSchema.GetDDL(ddl.Config{Comments: true, ProtectIds: false, Tables: true, ForeignKeys: true})
+	if len(spddl) == 0 {
+		spddl = []string{"\n-- Schema is empty -- no tables found\n"}
 	}
 	l := []string{
 		fmt.Sprintf("-- Schema generated %s\n", now.Format("2006-01-02 15:04:05")),
-		strings.Join(ddl, ";\n\n"),
+		strings.Join(spddl, ";\n\n"),
 		"\n",
 	}
 	if _, err := f.WriteString(strings.Join(l, "")); err != nil {
 		fmt.Fprintf(out, "Can't write out schema file: %v\n", err)
 		return
 	}
 	fmt.Fprintf(out, "Wrote schema to file '%s'.\n", name)
+
+	/*
+		The schema file written above includes comments, includes foreign keys
+		and doesn't add backticks around table and column names.
+		This file is intended for explanatory and documentation purposes, and
+		is not strictly legal Cloud Spanner DDL (Cloud Spanner doesn't currently support comments).
+
+		We change 'Comments' to false and 'ProtectIds' to true below to write out a
+		schema file that is a legal Cloud Spanner DDL.
+	*/
+	name = strings.Replace(name, ".txt", ".ddl.txt", 1)
```

Comment:
maybe nitpicking, but is it possible to only replace the last instance of ".txt" with ".ddl.txt"?